pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -1058,23 +951,11 @@ func (l *LibvirtDomainManager) migrate(vmi *v1.VirtualMachineInstance, options *
 	log.Log.Object(vmi).Infof("Live migration succeeded.")
 }
 
-func (l *LibvirtDomainManager) updateVMIMigrationMode(dom cli.VirDomain, vmi *v1.VirtualMachineInstance, mode v1.MigrationMode) error {
-	domainSpec, err := l.getDomainSpec(dom)
-	if err != nil {
-		return err
-	}
-
-	if domainSpec.Metadata.KubeVirt.Migration == nil {
-		domainSpec.Metadata.KubeVirt.Migration = &api.MigrationMetadata{}
-	}
-
-	domainSpec.Metadata.KubeVirt.Migration.Mode = mode
-
-	d, err := l.setDomainSpecWithHooks(vmi, domainSpec)
-	if err != nil {
-		return err
-	}
-	defer d.Free()
-
-	return nil
+func (l *LibvirtDomainManager) updateVMIMigrationMode(mode v1.MigrationMode) {
+	l.metadataCache.Migration.WithSafeBlock(func(migrationMetadata *api.MigrationMetadata, _ bool) {
+		if migrationMetadata.Mode != mode {
+			migrationMetadata.Mode = mode
+			l.metadataCache.Notify()
```

Comment:
another place where notify is called within the mutex protected function. Technically, it's not a huge deal or anything, i just don't want people to think that it's okay to do anything within this function other than set data on the metadata.